server/openapi_server/controllers/teacher_controller.py

```
@@ -0,0 +1,79 @@
+        if result.get("acknowledged"):
+            teacher.id = str(result.get("inserted_id"))  # Set the generated ID in the Teacher object
+            return teacher
+        raise HTTPException(status_code=500, detail="Failed to create teacher")
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=str(e))
+
+@teachers_router.put("/{id}", response_model=Teacher)
+def update_teacher(id: str, teacher: Teacher):
+    """Update a teacher's details."""
+    # Validate ID format
+    if not ObjectId.is_valid(id):
```

Ruff (0.11.9)

Undefined name `ObjectId`

(F821)